xbmc/pvr/PVRGUIDirectory.cpp

```diff
@@ -0,0 +1,504 @@
+    bool bShowHiddenChannels = StringUtils::EndsWithNoCase(fileName, ".hidden");
+    if (bShowHiddenChannels || strGroupName == "*") // all channels
+      group = CServiceBroker::GetPVRManager().ChannelGroups()->GetGroupAllRadio();
+    else
+      group = CServiceBroker::GetPVRManager().ChannelGroups()->GetRadio()->GetByName(strGroupName);
+
+    if (group)
+    {
+      const std::vector<PVRChannelGroupMember> groupMembers = group->GetMembers();
+      for (const auto& groupMember : groupMembers)
+      {
+        if (bShowHiddenChannels != groupMember.channel->IsHidden())
+          continue;
+
+        results.Add(std::make_shared<CFileItem>(groupMember.channel));
+      }
+    }
+    else
+    {
+      CLog::LogF(LOGERROR, "Unable to obtain members of channel group '%s'", strGroupName.c_str());
+      return false;
+    }
+
+    FilterDirectory(results);
+    return true;
+  }
+
+  return false;
+}
+
+namespace
+{
+  bool GetTimersRootDirectory(const CPVRTimersPath& path,
+                              const std::vector<std::shared_ptr<CPVRTimerInfoTag>>& timers,
+                              CFileItemList& results)
+  {
+    std::shared_ptr<CFileItem> item(new CFileItem(CPVRTimersPath::PATH_ADDTIMER, false));
+    item->SetLabel(g_localizeStrings.Get(19026)); // "Add timer..."
+    item->SetLabelPreformatted(true);
+    item->SetSpecialSort(SortSpecialOnTop);
+    item->SetIconImage("DefaultTVShows.png");
+    results.Add(item);
+
+    bool bRadio = path.IsRadio();
+    bool bRules = path.IsRules();
+
+    bool bHideDisabled = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(CSettings::SETTING_PVRTIMERS_HIDEDISABLEDTIMERS);
+
+    for (const auto& timer : timers)
+    {
+      if ((bRadio == timer->m_bIsRadio || (bRules && timer->m_iClientChannelUid == PVR_TIMER_ANY_CHANNEL)) &&
+          (bRules == timer->IsTimerRule()) &&
+          (!bHideDisabled || (timer->m_state != PVR_TIMER_STATE_DISABLED)))
+      {
+        item.reset(new CFileItem(timer));
+        const CPVRTimersPath timersPath(path.GetPath(), timer->m_iClientId, timer->m_iClientIndex);
+        item->SetPath(timersPath.GetPath());
+        results.Add(item);
+      }
+    }
+    return true;
+  }
+
+  bool GetTimersSubDirectory(const CPVRTimersPath& path,
+                             const std::vector<std::shared_ptr<CPVRTimerInfoTag>>& timers,
+                             CFileItemList& results)
+  {
+    bool bRadio = path.IsRadio();
+    int iParentId = path.GetParentId();
+    int iClientId = path.GetClientId();
+
+    bool bHideDisabled = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(CSettings::SETTING_PVRTIMERS_HIDEDISABLEDTIMERS);
+
+    std::shared_ptr<CFileItem> item;
+
+    for (const auto& timer : timers)
+    {
+      if ((timer->m_bIsRadio == bRadio) &&
+          (timer->m_iParentClientIndex != PVR_TIMER_NO_PARENT) &&
+          (timer->m_iClientId == iClientId) &&
+          (timer->m_iParentClientIndex == iParentId) &&
```

Comment:
> `warning: comparison between signed and unsigned integer expressions [-Wsign-compare]`

Reply:
Yeah, that's why the jenkins build fails. I will fix that, ofc.